src/instance.rs

```diff
@@ -71,55 +209,82 @@ where
     T: PyTypeInfo,
 {
     /// Borrows a GIL-bound reference to the contained `T`. By binding to the GIL lifetime, this
-    /// allows the GIL-bound reference to not require `Python` for any of its methods.
+    /// allows the GIL-bound reference to not require [`Python<'py>`](crate::Python) for any of its methods,
+    /// which makes calling methods on it more ergonomic.
     ///
     /// For native types, this reference is `&T`. For pyclasses, this is `&PyCell<T>`.
     ///
+    /// Note that the lifetime of the returned reference is the shortest of `&self` and
+    /// [`Python<'py>`](crate::Python).
+    /// Consider using [`Py::into_ref`] instead if this poses a problem.
+    ///
     /// # Examples
+    ///
     /// Get access to `&PyList` from `Py<PyList>`:
     ///
     /// ```
     /// # use pyo3::prelude::*;
     /// # use pyo3::types::PyList;
-    /// # Python::with_gil(|py| {
-    /// let list: Py<PyList> = PyList::empty(py).into();
-    /// let list: &PyList = list.as_ref(py);
-    /// assert_eq!(list.len(), 0);
-    /// # });
+    /// #
+    /// Python::with_gil(|py| {
```

Comment:
why make the `with_gil` visible?  Where the `py` token comes from is accidental, it could just as well be within a `#[pymethods]` method that gets it as an argument.

Reply:
I think it's weird to see an example conjuring variables out of nowhere. It also makes the example harder to copy-paste.

Reply:
I never understood this argument - next you'll have to supply a function around it, and finally a whole project including `Cargo.toml`.  It is very well understood what `py` is in the context of PyO3.

People are not supposed to blindly copy-paste code examples, they're supposed to understand what they show and apply it to their situation.  (end of rant)